packages/zowe-explorer/src/trees/job/JobInit.ts

```diff
@@ -161,9 +161,10 @@ export class JobInit {
             })
         );
 
-        const paginationEnabled = SettingsConfig.getDirectValue<boolean>("zowe.jobs.settings.pagination");
-        if (paginationEnabled) {
-            context.subscriptions.push(vscode.commands.registerCommand("zowe.jobs.loadMoreRecords", (doc) => JobActions.loadMoreRecords(doc)));
+        const paginationEnabled = SettingsConfig.getDirectValue<boolean>("zowe.jobs.paginate.enabled");
+        const recordsToFetch = SettingsConfig.getDirectValue<number>("zowe.jobs.paginate.recordsToFetch");
+        if (paginationEnabled && recordsToFetch > 1) {
+            context.subscriptions.push(vscode.commands.registerCommand("zowe.jobs.loadMoreRecords", (uri) => JobActions.loadMoreRecords(uri)));
             vscode.languages.registerCodeLensProvider(
                 { scheme: ZoweScheme.Jobs },
                 new PaginationCodeLens("zowe.jobs.loadMoreRecords", (doc) => JobFSProvider.instance.supportSpoolPagination(doc))
```

Comment:
I'm seeing the same behavior as Adam. As part of resolving this, I think we should update `JobFSProvider.supportSpoolPagination` so that it returns false if pagination is disabled, so that the code lens is hidden in the editor when its off.